File: apple/Sources/CCDAEngine/Models/CCDAPatient.swift
import Foundation

public struct CCDAPatient {
    public let ids: [CCDAIdentifier]
    public let name: CCDAHumanName?
    public let gender: CCDACodedValue?
    public let birthTime: String?
    public let maritalStatus: CCDACodedValue?
    public let race: CCDACodedValue?
    public let ethnicity: CCDACodedValue?
    public let addresses: [CCDAAddress]
    public let telecoms: [String]
}

File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Patient/CCDAPatient.kt
package com.ccdakit.engine

/**
 * Patient demographics parsed from `recordTarget/patientRole`.
 *
 * @property ids Patient identifiers.
 * @property name Patient name.
 * @property gender Administrative gender code.
 * @property birthTime Birth timestamp.
 * @property maritalStatus Marital status code.
 * @property race Race code.
 * @property ethnicity Ethnicity code.
 * @property addresses Postal addresses.
 * @property telecoms Telecom values such as phone or email URIs.
 */
data class CCDAPatient(
    val ids: List<CCDAIdentifier> = emptyList(),
    val name: CCDAHumanName? = null,
    val gender: CCDACodedValue? = null,
    val birthTime: CCDATimestamp? = null,
    val maritalStatus: CCDACodedValue? = null,
    val race: CCDACodedValue? = null,
    val ethnicity: CCDACodedValue? = null,
    val addresses: List<CCDAAddress> = emptyList(),
    val telecoms: List<String> = emptyList(),
)
